controllers/rhmi/rhmi_controller.go

```diff
@@ -371,12 +371,18 @@ func (r *RHMIReconciler) Reconcile(ctx context.Context, request ctrl.Request) (c
 	// either not checked, or rechecking preflight checks
 	if installation.Status.PreflightStatus == rhmiv1alpha1.PreflightInProgress ||
 		installation.Status.PreflightStatus == rhmiv1alpha1.PreflightFail {
-		return r.preflightChecks(installation, installType, configManager)
+		err := r.preflightChecks(installation, installType, configManager)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
 	}
 
 	// If the CR is being deleted, handle uninstall and return
 	if installation.DeletionTimestamp != nil {
-		return r.handleUninstall(installation, installType, request)
+		err := r.handleUninstall(installation, installType, request)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
```

Comment:
```suggestion
		err := r.handleUninstall(installation, installType, request)
		if err != nil {
			return ctrl.Result{}, err
		}
		return retryRequeue, nil
```

Should we return if `err` is `nil`  also ? Otherwise I think the installation logic will be run during uninstallation 🤔

Reply:
Doesn't look to affect uninstallation though, so maybe not 🤔

Reply:
Hmm, actually yes, the controller will not be requeued when `handleUninstall` returns nill, make sense to add `retryRequeue`. https://github.com/integr8ly/integreatly-operator/blob/bc737b41ae13ee7fdbfceb29abb3fe874bc63b46/controllers/rhmi/rhmi_controller.go#L374-L377 there as well..
Thanks for spotting it!

Reply:
> Hmm, actually yes, the controller will not be requeued when `handleUninstall` returns nill, make sense to add `retryRequeue`.
> 
> https://github.com/integr8ly/integreatly-operator/blob/bc737b41ae13ee7fdbfceb29abb3fe874bc63b46/controllers/rhmi/rhmi_controller.go#L374-L377
> 
> there as well..
> Thanks for spotting it!

Yes, I think it should be requeued at preflight checks on nil error also 👍